src/commands/force/source/delete.ts

```diff
@@ -0,0 +1,206 @@
+
+export class Delete extends DeployCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessage('examples').split(os.EOL);
+  public static readonly requiresProject = true;
+  public static readonly requiresUsername = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    checkonly: flags.boolean({
+      char: 'c',
+      description: messages.getMessage('flags.checkonly'),
+      longDescription: messages.getMessage('flags.checkonlyLong'),
+    }),
+    wait: flags.minutes({
+      char: 'w',
+      default: Duration.minutes(Delete.DEFAULT_SRC_WAIT_MINUTES),
+      min: Duration.minutes(1),
+      description: messages.getMessage('flags.wait'),
+    }),
+    testlevel: flags.enum({
+      char: 'l',
+      description: messages.getMessage('flags.testLevel'),
+      options: ['NoTestRun', 'RunLocalTests', 'RunAllTestsInOrg'],
+      default: 'NoTestRun',
+    }),
+    noprompt: flags.boolean({
+      char: 'r',
+      description: messages.getMessage('flags.noprompt'),
+    }),
+    metadata: flags.array({
+      char: 'm',
+      description: messages.getMessage('flags.metadata'),
+      exclusive: ['manifest', 'sourcepath'],
+    }),
+    sourcepath: flags.array({
+      char: 'p',
+      description: messages.getMessage('flags.sourcepath'),
+      longDescription: messages.getMessage('flags.sourcepathLong'),
+      exclusive: ['manifest', 'metadata'],
+    }),
+    verbose: flags.builtin({
+      description: messages.getMessage('flags.verbose'),
+    }),
+  };
+  protected xorFlags = ['metadata', 'sourcepath'];
+  protected readonly lifecycleEventNames = ['predeploy', 'postdeploy'];
+  private sourceComponents: SourceComponent[];
+  private isRest = false;
+
+  private updateDeployId = once((id) => {
+    this.displayDeployId(id);
+    this.setStash(id);
+  });
+
+  public async run(): Promise<DeployCommandResult> {
+    await this.delete();
+    this.resolveSuccess();
+    return this.formatResult();
+  }
+
+  protected async delete(): Promise<void> {
+    // verify that the user defined one of: metadata, sourcepath
+    this.validateFlags();
+
+    this.componentSet = await ComponentSetBuilder.build({
+      apiversion: this.getFlag<string>('apiversion'),
+      sourceapiversion: await this.getSourceApiVersion(),
+      sourcepath: this.getFlag<string[]>('sourcepath'),
+      metadata: this.flags.metadata && {
+        metadataEntries: this.getFlag<string[]>('metadata'),
+        directoryPaths: this.getPackageDirs(),
+      },
+    });
+
+    this.sourceComponents = this.componentSet.getSourceComponents().toArray();
+
+    if (!this.sourceComponents.length) {
+      // if we didn't find any components to delete, let the user know and exit
+      // matches toolbelt
+      this.ux.styledHeader(chalk.blue('Deleted Source'));
+      this.ux.log('No results found');
+      this.exit(0);
+    }
```

Comment:
I think we should keep formatting to the formatters unless it's not possible.  Also, we should avoid calling exit explicitly and let the command framework handle it.  What if we just did `return;` here and let the formatter display this?